api/src/test/java/org/openmrs/validator/DiagnosisValidatorTest.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.validator;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.CodedOrFreeText;
+import org.openmrs.ConditionVerificationStatus;
+import org.openmrs.Diagnosis;
+import org.openmrs.Encounter;
+import org.openmrs.test.BaseContextSensitiveTest;
+import org.springframework.validation.BindException;
+import org.springframework.validation.Errors;
+
+public class DiagnosisValidatorTest extends BaseContextSensitiveTest {
+
+	private Diagnosis diagnosis;
+	private Errors errors;
+
+	@Before
+	public void setUp() {
+		diagnosis = new Diagnosis();
+		errors = new BindException(diagnosis, "diagnosis");
+	}
+	
+	/**
+	 * @see DiagnosisValidator#validate(Object, Errors)
+	 */
+	@Test
+	public void validate_shouldFailValidationIfEncounterIsNull(){
+		diagnosis.setEncounter(null);
+		new DiagnosisValidator().validate(diagnosis, errors);
+		Assert.assertTrue(errors.hasFieldErrors("encounter"));
+	}
+	@Test
+	public void validate_shouldFailValidationIfCodedOrFreeTextIsNull(){
```

Comment:
These tests would be easier to understand if you used the methods names instead of data types.
For instance validate_shouldFailValidationIfConditionIsNull is easier than validate_shouldFailValidationIfCodedOrFreeTextIsNull

Can you apply the same patter all over?